examples/notebooks/integrate_3rd_transforms.ipynb

```diff
@@ -0,0 +1,294 @@
+    "    CropForegroundd(keys=['image', 'label'], source_key='image'),\n",
+    "    # add another dim as BatchGenerator and Rising expects shape [B, C, H, W, D]\n",
+    "    AddChanneld(keys=['image', 'label'])\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup BatchGenerator transforms\n",
+    "Note:\n",
+    "1. BatchGenerator requires the arg is `**data`, can't compose with MONAI transforms directly, need `adaptor`.\n",
+    "2. BatchGenerator requires data shape is [B, C, H, W, D], MONAI requires [C, H, W, D]."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "batch_generator_transforms = ContrastAugmentationTransform(data_key=\"image\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup TorchIO transforms\n",
+    "Note:\n",
+    "1. TorchIO specifies several keys internally, use `adaptor` if conflicts.\n",
+    "2. There are few example or tutorial, hard to quickly get start.\n",
+    "3. The TorchIO transforms depend on many TorchIO modules(Subject, Image, ImageDataset, etc.), not easy to support MONAI dict input data.\n",
+    "4. It can handle PyTorch Tensor data(shape: [C, H, W, D]) directly, so used it to handle Tensor in this tutorial.\n",
+    "5. If input data is Tensor, it can't support dict type, need `adaptor`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "torchio_transforms = RescaleIntensity(out_min_max=(0., 1.), percentiles=(0.05, 99.5))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup Rising transforms\n",
+    "Note:\n",
+    "1. Rising inherits from PyTorch `nn.Module`, expected input data type is PyTorch Tensor, so can only work after `ToTensor`.\n",
+    "2. Rising requires data shape is [B, C, H, W, D], MONAI requires [C, H, W, D].\n",
+    "3. Rising requires the arg is `**data`, need `adaptor`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rising_transforms = Mirror(dims=DiscreteParameter((0, 1, 2)), keys=[\"image\", \"label\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Compose all transforms"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "transform = Compose(monai_transforms + [\n",
+    "    adaptor(batch_generator_transforms, {'image': 'image'}),\n",
+    "    ToTensord(keys=['image', 'label']),\n",
+    "    adaptor(rising_transforms, {'image': 'image', 'label': 'label'}),\n",
+    "    adaptor(lambda **data: (data['image'][0], data['label'][0]), ['image', 'label']),\n",
+    "    adaptor(lambda **data: torchio_transforms(data['image']), 'image')\n",
```

Comment:
this could be: `adaptor(torchio_transforms, ['image'], {'image': "data"})`

Reply:
updated in the latest commit.
Thanks.